### ISSUE TITLE
Can it do bootstrapping? It is so slow.

### ISSUE THREAD

**Issue by zaoqi-unsafe** [andrei-markeev/ts2c#35]

**Comment by andrei-markeev** [comment]
Not sure what you mean by _bootstrapping_, but big part of transpilation is preprocessing provided by TypeScript (and I cannot do anything about it).
If you could provide some concrete examples where transpilation is very slow in TS2C comparing to normal TypeScript (e.g. compare `time tsc test.ts` and `time ts2c test.ts`), then I might be able to do some optimizations, even though I don't really consider transpilation speed important at this point.

**Comment by zaoqi** [comment]
https://github.com/andrei-markeev/ts2c/blob/master/package.json

2.8.3 is pretty old.

**Comment by zaoqi** [comment]
Enum is not implemented.So I need to first `tsc`.

**Comment by andrei-markeev** [comment]
> 2.8.3 is pretty old

true, but for now I am only implementing ES3 spec, so it doesn't really matter which TS version is used

> Enum is not implemented. So I need furst `tsc`

My previous comment still stands: `ts2c` sort of runs `tsc` behind the scenes for getting AST and semantic information. So running `ts2c` cannot be faster than `tsc` on the same file.

**Comment by andrei-markeev** [comment]
Small update: I am prioritizing compilation performance now. It will be improved. The current plan is to get rid of TypeScript dependency and use a more lightweight and faster parser instead. Also, type inference mechanism of TS2C should probably be revorked.

**Comment by andrei-markeev** [comment]
Migration to a faster parser completed as part of https://github.com/andrei-markeev/ts2c/pull/101
